samza-core/src/main/java/org/apache/samza/zk/ZkBarrierForVersionUpgrade.java

```diff
@@ -162,18 +164,22 @@ public void handleChildChange(String parentPath, List<String> currentChildren) {
    * Barrier state values are either DONE or TIMED_OUT. It only registers to receive on valid state change notification.
    * Once a valid state change notification is received, it will un-subscribe from further notifications.
    */
-  class ZkBarrierReachedHandler implements IZkDataListener {
+  class ZkBarrierReachedHandler extends ZkUtils.GenIZkDataListener {
     private final String barrierStatePath;
     private final String barrierVersion;
 
-    public ZkBarrierReachedHandler(String barrierStatePath, String version) {
+    public ZkBarrierReachedHandler(String barrierStatePath, String version, ZkUtils zkUtils) {
+      super(zkUtils, "ZkBarrierReachedHandler");
       this.barrierStatePath = barrierStatePath;
       this.barrierVersion = version;
     }
 
     @Override
     public void handleDataChange(String dataPath, Object data) {
       LOG.info("got notification about barrier " + barrierStatePath + "; done=" + data);
+      if (notAValidEven())
```

Comment:
typo fix done. thanks.
I want to keep it at the info level for at least first release, in case we need to debug it.
For the next version we will review log levels to minimize logging.